### ISSUE TITLE
[Offline] Connect local mutations and changeset records to GraphQL soil data push mutation

### ISSUE THREAD

**Issue by tm-ruxandra** [techmatters/terraso-mobile-client#2279]
Combine the work done in #2277, #2278, and [#1485](https://github.com/techmatters/terraso-backend/issues/1485). (We are now referring to this as the "push" mutation).

Create a mechanism to handle soil data mutations with their local equivalents and sync changes through the push mutation when the app is a) online and b) has unsynced changes within a 500ms debounce window. `ChangeSet` records (or whatever they end up being called) should be updated on a successful push, unless there are new changes that have come in since the push was started. (In this case, treat the operation as if it hadn't occurred, and allow the system to attempt to push again). Any `SoilData` which was pushed should be updated to match the server's response.

If the push operation itself fails, retry after some time period. Discard any changes that resulted in a business error.

All of this functionality should be gated behind the offline functionality feature flag; if the flag is disabled, the app should continue to operate as it currently does.

Reference the prototype implementation in the `feat/local-storage-sync-prototype` branch.

**Comment by tm-ruxandra** [comment]
Blocked on https://github.com/techmatters/terraso-backend/issues/1485; cannot do full testing until backend system comes online.

**Comment by DerekCaelin** [comment]
@tm-ruxandra was this the issue we thought we could postpone QA until other issues are finished?

**Comment by tm-ruxandra** [comment]
@DerekCaelin No, this issue is ready for QA. You are thinking of #2278.

**Comment by tm-ruxandra** [comment]
@DerekCaelin Yeah, turn on the feature flag and enter data both with network connectivity online and offline. You should see all data persist on the server when your device returns to online mode.

**Comment by DerekCaelin** [comment]
In testing with @CourtneyLee333 this afternoon, we entered data while online and offline. We checked whether our data persisted when we came online, and it did (unless someone entered competing information and pushed after, which is expected). This looks great.
